service/history/nDCHistoryReplicator.go

```diff
@@ -866,18 +866,18 @@ func (r *nDCHistoryReplicatorImpl) backfillHistory(
 		lastEventID,
 		lastEventVersion),
 	)
-	var lastHistoryBatch *commonpb.DataBlob
-	prevTxnID := common.EmptyVersion
 	historyBranch, err := serialization.HistoryBranchFromBlob(branchToken, enumspb.ENCODING_TYPE_PROTO3.String())
 	if err != nil {
 		return nil, err
 	}
-	latestBranchID := historyBranch.GetBranchId()
-	var prevBranchID string
 
-	sortedAncestors := copyAndSortAncestors(historyBranch.GetAncestors())
+	prevTxnID := common.EmptyVersion
+	var lastHistoryBatch *commonpb.DataBlob
+	var prevBranchID string
+	sortAncestors(historyBranch.GetAncestors())
+	sortedAncestors := historyBranch.GetAncestors()
```

Comment:
I would add 
```
return ans
```
to `sortAncestors()` and replace this 2 lines with:

```suggestion
	sortedAncestors := sortAncestors(historyBranch.GetAncestors())
```

Reply:
updated